### PR TITLE
Add classification row export model

### PR DESCRIPTION
**Pull request by camallen**
New table to support pre-computed export csv data and populate the data in the classification lifecycle event via a low data worker

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: db/migrate/20171019115705_add_classification_export_row_model.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    create_table :classification_export_rows do |t|
+      t.references :classification, null: false
+      t.references :project, null: false
+      t.references :workflow, null: false, index: true
+      t.references :user
+      t.string :user_name
+      t.string :user_ip
+      t.string :workflow_name
+      t.string :workflow_version
+      t.timestamp :created_at
```
Comment (camallen) [inline db/migrate/20171019115705_add_classification_export_row_model.rb]: this needs to be changed to the `classification.created_at`

File: db/migrate/20171019115705_add_classification_export_row_model.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      t.jsonb :annotations
+      t.jsonb :subject_data
+      t.string :subject_ids
+      t.timestamps
+    end
+
+    add_index :classification_export_rows,
+      %i(project_id classification_id),
+      name: 'index_export_rows_on_project_id_and_classification_id',
+      unique: true
```
Comment (marten) [inline db/migrate/20171019115705_add_classification_export_row_model.rb]: Aren't classification IDs globally unique? Why include the project in this unique index?

File: lib/classification_lifecycle.rb [hunk 3]
```diff
@@ -86,6 +89,11 @@ def notify_cellect
     end
   end
 
+  def create_export_row
```
Comment (marten) [inline lib/classification_lifecycle.rb]: Do we want a feature flag on this?

Comment (camallen) [reply lib/classification_lifecycle.rb]: Yeah - good call.

File: db/migrate/20171019115705_add_classification_export_row_model.rb (hunk 2 above)
Comment (camallen) [reply db/migrate/20171019115705_add_classification_export_row_model.rb]: they are unique, i want the project index to increase how selective this index when we run a dump for backfilling. Finding can probably just use the project_id / workflow_id index but i should check the `find_in_batches` won't use the compound index to order by classification (i think it does by default).

File: lib/classification_lifecycle.rb [hunk 4]
```diff
@@ -86,6 +89,13 @@ def notify_cellect
...
+    return unless classification.complete?
+    unless Panoptes.flipper[:disable_classification_export_row_in_lifecycle].enabled?
```
Comment (marten) [inline lib/classification_lifecycle.rb]: The negatives is a bit confusing here.

**Pull request merged**